### DISCUSSION TITLE
How can I use subtags?

### DISCUSSION THREAD

**Discussion by kenji-webdev** [amannn/next-intl#471] [Q&A]
I've tried using both en-US and pt-BR locales but I can't get them to work, I'm just redirected to `/en`, and if I try to navigate directly to, for example, `pt-BR`, I end up in `/en/pt-BR`

this is my `src` folder

![image](https://github.com/amannn/next-intl/assets/98567681/88eb9355-f2db-4970-afef-f1492cd8b591)

I've renamed the `message/*.json` files,

the values inside the locale property,
```
export async function generateStaticParams() {
  return [{ locale: "en" }, { locale: "pt-BR" }];
}
```
the `middleware.ts`.
```
import createMiddleware from "next-intl/middleware";

export default createMiddleware({
  // A list of all locales that are supported
  locales: ["en", "pt-BR"],

  // If this locale is matched, pathnames work without a prefix (e.g. `/about`)
  defaultLocale: "pt-BR",
});

export const config = {
  // Skip all paths that should not be internationalized. This example skips the
  // folders "api", "_next" and all files with an extension (e.g. favicon.ico)
  matcher: ["/((?!api|_next|.*\\..*).*)"],
};
```

I've also tried using an underline instead of a dash but that didn't work either, 

The keywords I've used to search for previous discussions before posting this one were: "subtags", "subtag", "pt-BR" and "en-US", I've removed the `is:open` tag as well.

**Comment by amannn** [comment]
I just tried to reproduce this in a CodeSandbox: https://codesandbox.io/p/sandbox/next-intl-bug-template-app-forked-kpgfsy?file=/messages/pt-BR.json:3,30

When I open https://kpgfsy-3000.csb.app/pt-BR, I see in my devtools:

<img width="332" alt="Screenshot 2023-08-23 at 17 40 53" src="https://github.com/amannn/next-intl/assets/4038316/633ba3f2-6380-4512-8009-231dfcac1cfd">

… and then "Home from pt-BR" renders for me.

Anything you can spot that you're doing differently?

The middleware has quite an extensive test suite in case you find a particular request that you'd like to debug: https://github.com/amannn/next-intl/blob/main/packages/next-intl/test/middleware/middleware.test.tsx

**Reply by kenji-webdev** [reply]
Hey, I did as it was shown in the CodeSandbox, I think I might've had a local issue, when I tested my repository at the work pc it was running as expected.

As for the network requests, I noticed that when I visited the route, I was making a request like the one in your screenshot, but in my case I was also being redirected to a lower-cased route that then continuously called itself and returned `404's`

Sorry, I'm not sure of what exactly caused the error, but it stopped happening in my computer after I updated chrome to the latest version as of now, `Version 116.0.5845.111` for `64bits`, sorry I didn't keep track of exactly which version I was using previously, but it wasn't an old one, as I remember it was also part of `116.X`

Thank you for taking the time to answer this discussion and creating a test

**Reply by amannn** [reply]
Ok sure, glad to hear it works now!
